zhaquirks/ikea/fivebtnremotezha.py

```diff
@@ -111,77 +100,13 @@ class IkeaTradfriRemote1(CustomDevice):
     }
 
     device_automation_triggers = {
-        (SHORT_PRESS, TURN_ON): {
-            COMMAND: COMMAND_TOGGLE,
-            CLUSTER_ID: 6,
-            ENDPOINT_ID: 1,
-        },
-        (LONG_PRESS, TURN_ON): {
-            COMMAND: COMMAND_RELEASE,
-            CLUSTER_ID: 5,
-            ENDPOINT_ID: 1,
-            PARAMS: {"param1": 537},
-        },
-        (SHORT_PRESS, DIM_UP): {
-            COMMAND: COMMAND_STEP_ON_OFF,
-            CLUSTER_ID: 8,
-            ENDPOINT_ID: 1,
-            PARAMS: {"step_mode": 0},
-        },
-        (LONG_PRESS, DIM_UP): {
-            COMMAND: COMMAND_MOVE_ON_OFF,
-            CLUSTER_ID: 8,
-            ENDPOINT_ID: 1,
-            PARAMS: {"move_mode": 0},
-        },
-        (SHORT_PRESS, DIM_DOWN): {
-            COMMAND: COMMAND_STEP,
-            CLUSTER_ID: 8,
-            ENDPOINT_ID: 1,
-            PARAMS: {"step_mode": 1},
-        },
-        (LONG_PRESS, DIM_DOWN): {
-            COMMAND: COMMAND_MOVE,
-            CLUSTER_ID: 8,
-            ENDPOINT_ID: 1,
-            PARAMS: {"move_mode": 1},
-        },
-        (SHORT_PRESS, LEFT): {
-            COMMAND: COMMAND_PRESS,
-            CLUSTER_ID: 5,
-            ENDPOINT_ID: 1,
-            PARAMS: {
-                "param1": 257,
-                "param2": 13,
-                "param3": 0,
-            },
-        },
-        (LONG_PRESS, LEFT): {
-            COMMAND: COMMAND_HOLD,
-            CLUSTER_ID: 5,
-            ENDPOINT_ID: 1,
-            PARAMS: {
-                "param1": 3329,
-                "param2": 0,
-            },
-        },
-        (SHORT_PRESS, RIGHT): {
-            COMMAND: COMMAND_PRESS,
-            CLUSTER_ID: 5,
-            ENDPOINT_ID: 1,
-            PARAMS: {
-                "param1": 256,
-                "param2": 13,
-                "param3": 0,
-            },
-        },
-        (LONG_PRESS, RIGHT): {
-            COMMAND: COMMAND_HOLD,
-            CLUSTER_ID: 5,
-            ENDPOINT_ID: 1,
-            PARAMS: {
-                "param1": 3328,
-                "param2": 0,
+        **IkeaTradfriRemote.device_automation_triggers,
+        **{
+            (LONG_PRESS, TURN_ON): {
+                COMMAND: COMMAND_RELEASE,
+                CLUSTER_ID: 5,
+                ENDPOINT_ID: 1,
+                PARAMS: {"param1": 537},
             },
```

Comment:
Hmm, so only that "firmware" uses a different parameter for LONG_PRESS/ON?
The code itself looks good to me (and copies previous behavior). Is there any thread on this or can someone confirm that one "version" actually uses this parameter? 

instead of:
 
```python
        (LONG_PRESS, TURN_ON): {
            COMMAND: COMMAND_RELEASE,
            CLUSTER_ID: 5,
            ENDPOINT_ID: 1,
            PARAMS: {"param1": 0},
        },
```

Reply:
I was getting problem with my remote and must doing one flash erase and writing one new bootloader main flash and user data and its have booted and is trying updating the firmware but its taking some hours.

The different parms is real for HA firmware is was 2 or more version and perhaps one with one mixed. The 2.3.080 was doing all the same and it was working OK.
Also the stop_with_on_off / stop is only working on 2.3.080 all earlier is using the same commands for doe both stop.

Then OTA updating firmware and the device is changing clusters on the endpoints is deleting and readding needed or the quirk is not working OK or not loaded.

Testing ZLL firmware i need finding one very old one (version 1.X) and SWD flashing it after extracting it with zigpy but i dont knowing if its making some sense then its old buggy one and its only updating the firmware and its working for the user.

I was thinking deleting all old version and only having the 2.3.080 with group binding and the latest with device binding but i think the maintainers dont like it.

Only is that the 2.3.080 is the last with working group binding for user that is using Zigbee Light Groups.

Reply:
So in your opinion, the changes should be fine, right?

> I was thinking deleting all old version and only having the 2.3.080 with group binding and the latest with device binding but i think the maintainers dont like it.

Yeah, we shouldn't really do this, as (1) many users don't update their devices and (2) ZHA might still have an outdated signature cached.

> Only is that the 2.3.080 is the last with working group binding for user that is using Zigbee Light Groups.

Hmm, I remember you posted a table with the corresponding firmware versions for group binding (for every device). Can't find it now, can you link me that again?

https://www.zigbee2mqtt.io/devices/E1524_E1810.html mentions 2.3.075 and later only supports device binding.

Reply:
> Hmm, so only that "firmware" uses a different parameter for LONG_PRESS/ON?

I think this is probably a mistake: The `release` event is also used for the arrows release, and in this case `param1` holds the time it stayed pressed. Someone probably saw this event and put it as `LONG_PRESS/ON` ?

 Also I think we should remove `LONG_PRESS/ON` since it is not really working: you first get the `toggle` event anyway, and there is no way to tell if it's still holding or not. Last version (2.4.45) also seems to have remove it altogether.
More info regarding that [here](https://www.zigbee2mqtt.io/devices/E1524_E1810.html#toggle-button).

Instead, I think we should provide a generic event for release, whatever is inside `param1`. Unfortunately there is no param to discriminate between a release of left or right arrow. I am not sure what is the best way to do so however.

Reply:
After having a quick look at the rest of the repo perhaps `(LONG_RELEASE, LONG_RELEASE)` or `(LONG_RELEASE, BUTTON)` would make sense as a generic `release` event for both arrows ?

I tried both and `(LONG_RELEASE, BUTTON)` looks less ugly in the UI for me.

Reply:
> I think this is probably a mistake: The release event is also used for the arrows release, and in this case param1 holds the time it stayed pressed

I'd think we should just remove that "special case", as it seems to invalid.

> Also I think we should remove LONG_PRESS/ON since it is not really working

We would break existing automations if we completely remove a device automation trigger though. Let's just leave it in for now and have this PR be (1) refactoring + (2) adding the new UP/DOWN long release triggers + (3) removing the invalid LONG_PRESS/ON with "param 537".
For more changes, a new PR should be opened.